packages/matter.js/src/protocol/interaction/InteractionClient.ts

```diff
@@ -99,23 +115,48 @@ export class SubscriptionClient implements ProtocolHandler<MatterController> {
         }
         listener(dataReport);
     }
+
+    async close() {
+        this.subscriptionListeners.clear();
+        this.subscriptionUpdateTimers.forEach(timer => timer.stop());
+        this.subscriptionUpdateTimers.clear();
+    }
 }
 
 export class InteractionClient {
-    private readonly subscriptionListeners = new Map<number, (dataReport: DataReport) => void>();
-    private readonly subscriptionUpdateTimers = new Map<number, Timer>();
     // TODO Add storage for these data to be able to optimize follow up subscriptions by only request updated data
     private readonly subscribedLocalValues = new Map<string, any>();
     private readonly subscribedClusterDataVersions = new Map<string, number>();
+    private readonly ownSubscriptionIds = new Set<number>();
+    private readonly subscriptionClient: SubscriptionClient;
 
     constructor(
         private readonly exchangeProvider: ExchangeProvider,
         readonly nodeId: NodeId,
     ) {
-        // TODO: Right now we potentially add multiple handlers for the same protocol, We need to fix this
-        this.exchangeProvider.addProtocolHandler(
-            new SubscriptionClient(this.subscriptionListeners, this.subscriptionUpdateTimers),
-        );
+        if (this.exchangeProvider.hasProtocolHandler(INTERACTION_PROTOCOL_ID)) {
+            const client = this.exchangeProvider.getProtocolHandler(INTERACTION_PROTOCOL_ID);
+            if (!(client instanceof SubscriptionClient)) {
+                throw new ImplementationError(
+                    `Unexpected protocol handler ${INTERACTION_PROTOCOL_ID} for InteractionClient already registered`,
```

Comment:
This is a little confusing.  Is it "Protocol handler ${INTERACTION_PROTOCOL_ID} client is not a SubscriptionClient"?

Reply:
yes it is this ... will rephrase